File: dart/167.dart
class Solution {
    List<int> twoSum(List<int> n, int t) {
        int s = 0, e = n.length - 1;
        while (s < e) {
            int y = n[s] + n[e];
            if (y == t)
                return [ s + 1, e + 1 ];
            if (y < t)
                s++;
            else
                e--;
        }
        return [];
    }
}

File: java/167.java
class Solution {
    public int[] twoSum(int[] n, int t) {
        int s = 0, e = n.length - 1;
        while (s < e) {
            if (n[s] + n[e] == t)
                return new int[] { s + 1, e + 1 };
            if (n[s] + n[e] < t)
                s++;
            else
                e--;
        }
        return null;
    }
}
